src/main/java/org/mule/runtime/api/metadata/ExpressionLanguageMetadataService.java

```
@@ -147,6 +160,16 @@ ModuleDefinition moduleDefinition(String nameIdentifier, Collection<ModuleDefini
       throws ExpressionCompilationException;
 
 
+  /**
+   * Returns a MetadataType from a given type expression.
+   *
+   * @param typeExpression The type expression to evaluate
+   * @param modules        Other modules available
+   * @return The resolved MetadataType
+   * @since 1.5
+   */
+  Optional<MetadataType> evaluateTypeExpression(String typeExpression, Collection<ModuleDefinition> modules);
```

This currently returns an Optional, so this can be tested from the mule side. In a next iteration we should work on throwing a descriptive error message instead of returning an optional.